src/components/maps/MapListComponent.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+    <div>
+        <div class="sort-options flex-row gap-md">
+            <SearchBox v-model="searchVal"/>
+            <div class="sort-dropdown">
+                <Select
+                    model-value="Name"
+                    v-model="sortMethod"
+                    :options="sortMethods"
+                    label="Sort By"
+                />
+            </div>
+
+        </div>
+        <div class="map-list">
+            <div class="maps">
+                <MapOverviewCard
+                    v-for="map in filteredMaps" :map="map"
+                    @mapSelected="mapSelected"
+                />
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
+
+/**
+ * TODO:
+ * - Similar to online map browser
+ * - Indicator whether map is installed or not
+ * - Easy one click install button
+ * - Demo map button that launches a simple offline game on the map
+ */
+
+import { computed, ref } from "vue";
+import Select from "@/components/controls/Select.vue";
+import MapOverviewCard from "@/components/maps/MapOverviewCard.vue";
+import SearchBox from "@/components/controls/SearchBox.vue";
+import { MapData } from "@/model/map-data";
+
+enum SortMethod {
+    Name = "Name",
+    Size = "Size",
+}
+
+const sortMethods = ref([SortMethod.Name, SortMethod.Size]);
+const sortMethod = ref(SortMethod.Name.valueOf());
+const searchVal = ref("");
+const emit = defineEmits(["mapSelected"]);
+const filteredMaps = computed(() => {
+    let maps = Array.from(api.content.maps.installedMaps);
+
+    if(searchVal.value.length > 0) {
+        maps = maps.filter((map: MapData) => {
+            return map.friendlyName.toLowerCase().includes(searchVal.value.toLowerCase());
+        })
+    }
+
+    switch (sortMethod.value) {
+        case SortMethod.Name:
+            maps.sort((a, b) => {
+                return a.friendlyName.localeCompare(b.friendlyName);
+            });
+            break;
+        case SortMethod.Size:
+            maps.sort((a, b) => {
+                return a.width * a.height - b.width * b.height;
+            })
+    }
+    return maps;
+});
+
+function mapSelected(map: MapData) {
+    emit("mapSelected", map);
```

Comment:
This emit name gets reused a lot. Should I make some kind of top-level event symbol instead of this hardcoded name in a few places?

Reply:
At the least, we can remove the `mapSelected` event on the `MapOverviewCard` element itself and instead add the `@click` handler in the parent component, done it here https://github.com/beyond-all-reason/bar-lobby/pull/85/commits/3a675db3c30d77a11a7d5779853d110286cbe368